### PR TITLE
Feat: add markdown emoji picker

### PR DESCRIPTION
**Pull request by tokitou-san**
**Tasks:**
 - [X] Add emoji picker via markdown logic
 - [ ] Add more efficient popover
 - [ ] Store emoji images locally
 - [ ] Show emoji image on popover

### **Current State**
Screenshot from 2023-06-04 16-10-25

### CODE REVIEW

File: src/lib/components/shared/text_editor.svelte [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+<div class="relative">
+    <textarea
+        on:input={input_handler}
+        bind:this={textarea_el}
+        class="h-[8vw] w-full rounded-[0.75vw] border-none bg-surface-900 p-[1vw] text-[1vw] leading-[1.5vw] text-surface-50 outline-none ring-2 ring-white/25 duration-300 ease-in-out placeholder:text-surface-200 focus:ring-2 focus:ring-white/50"
+        placeholder="Leave a comment"
+    />
+    <!-- Basic popover -->
+    {#if emoji_matching_list.length > 0}
```
Comment (baseplate-admin) [inline src/lib/components/shared/text_editor.svelte:47]: Cant we find a better implementation ( maybe an emoji tippy? )

File: src/lib/data/emojis.json [hunk 2]
```diff
@@ -0,0 +1,1879 @@
+{
+	"+1": "https://github.githubassets.com/images/icons/emoji/unicode/1f44d.png?v8",
```
Comment (baseplate-admin) [inline src/lib/data/emojis.json:2]: Dont hyperlink github. They will gut us

**Review by baseplate-admin — CHANGES_REQUESTED** [review]

File: src/lib/components/shared/text_editor.svelte (hunk 1 above)
Comment (tokitou-san) [reply src/lib/components/shared/text_editor.svelte:47]: Yea I already said - we will add better popover later ( now its a basic thing to show output )
Probably do once you approve above logic

File: src/lib/data/emojis.json (hunk 2 above)
Comment (tokitou-san) [reply src/lib/data/emojis.json:2]: Now using key instead, no call to github ?

File: src/lib/components/shared/text_editor.svelte (hunk 1 above)
Comment (baseplate-admin) [reply src/lib/components/shared/text_editor.svelte]: > Probably do once you approve above logic

I dont like this logic. But i will refractor it if needs to be

File: src/lib/data/emojis.json (hunk 2 above)
Comment (baseplate-admin) [reply src/lib/data/emojis.json:2]: > Now using key instead, no call to github ?

`https://github.githubassets.com/images/icons/emoji/unicode/1f44d.png?v8`

Images are still on github

**Review by baseplate-admin — CHANGES_REQUESTED** [review]

File: src/lib/data/emojis.json (hunk 2 above)
Comment (baseplate-admin) [reply src/lib/data/emojis.json:2]: > Now using key instead, no call to github ?

`https://github.githubassets.com/images/icons/emoji/unicode/1f44d.png?v8`
Images are still stored on github

**Comment by baseplate-admin** [comment]
Okay i see another way to do things. We should not use `emojis.json`. Instead embed everything inside the `text_editor.svelte` file

**Comment by tokitou-san** [comment]
@baseplate-admin It would be great if you can review this :D
Script used to download emojis:
```py
import os
import json
import requests

os.makedirs("emojis", exist_ok = True)

with open("emojis.json") as file:
	emoji_dict = json.load(file)

total_files = len(emoji_dict)

for i, (emoji_name, emoji_link) in enumerate(emoji_dict.items()):
	image_file_name = os.path.join("emojis", f"{emoji_name}.png")

	response = requests.get(emoji_link)

	with open(image_file_name, "wb") as image_file:
		image_file.write(response.content)

	files_left = total_files - i

	print(f"Downloaded emoji {emoji_name} to {image_file_name}")
	if files_left > 0:
		print(f"{files_left} files left")
```

**Comment by tokitou-san** [comment]
@baseplate-admin 
I added floating popover, but not completely finished.
Current issues:
 - showing empty element on invalid emoji_name
 - Its not working on new line

And also lmk if you want moving popover with caret ( now its fixed at start kinda like in github lol ).

**Pull request closed without merging**